app/controllers/NiphlNumberController.scala

```diff
@@ -56,15 +56,16 @@ class NiphlNumberController @Inject() (
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors))),
         niphlNumber => {
-          val updatedMaintainProfileAnswers =
-            request.userAnswers.maintainProfileAnswers.copy(niphlNumber = Some(NiphlNumber(niphlNumber)))
-          val updatedUserAnswers            = request.userAnswers.copy(maintainProfileAnswers = updatedMaintainProfileAnswers)
+          val updatedTgpModelObject =
+            request.userAnswers.maintainProfileAnswers
+              .copy(hasNiphl = Some(true), niphlNumber = Some(NiphlNumber(niphlNumber)))
```

Comment:
I don't think we should override hasNiphl here, it is possible that they will enter this value but never fill out the question page in which case we should fail with an error. We need a validation check when they submit in CYA to ensure there are no empty values for ukims and questions/answers for numbers they stated they have

Reply:
can we chat about this

Reply:
chatted - removed